src/pages/JobLogs.tsx

```diff
@@ -0,0 +1,162 @@
+import { Skeleton } from "antd";
+import { useParams } from "react-router-dom";
+import { useJobLogsAnalytics } from "analytics/joblogs/useJobLogsAnalytics";
+import { StyledRouterLink, StyledLink } from "components/styles";
+import {
+  getLobsterTestLogUrl,
+  getLobsterTestLogCompleteUrl,
+} from "constants/externalResources";
+import { getTaskRoute } from "constants/routes";
+import { useToastContext } from "context/toast";
+import {
+  GetDisplayTaskQuery,
+  GetDisplayTaskQueryVariables,
+  GetTestsQuery,
+  GetTestsQueryVariables,
+} from "gql/generated/types";
+import { GET_DISPLAY_TASK, GET_TESTS } from "gql/queries";
+import { PageDoesNotExist } from "pages/404";
+
+export const JobLogs = () => {
+  const dispatchToast = useToastContext();
+  const { sendEvent } = useJobLogsAnalytics();
+  const { taskId, execution: execStr, groupId } = useParams<{
+    taskId: string;
+    execution: string;
+    groupId: string;
+  }>();
+  const execution = parseInt(execStr, 10);
+  const {
+    data: displayTaskResult,
+    loading: isLoadingDisplayTask,
+    error: errorDisplayTask,
+  } = useQuery<GetDisplayTaskQuery, GetDisplayTaskQueryVariables>(
+    GET_DISPLAY_TASK,
+    {
+      variables: { taskId, execution },
+      onError: (err) =>
+        dispatchToast.error(
+          `There was an error loading the task: ${err.message}`
+        ),
+    }
+  );
+  const [
+    getTests,
+    { data: testsResult, loading: isLoadingTests, error: errorTests },
+  ] = useLazyQuery<GetTestsQuery, GetTestsQueryVariables>(GET_TESTS, {
+    onError: (err) =>
+      dispatchToast.error(
+        `There was an error loading test result data: ${err.message}`
+      ),
+  });
+
+  const task = displayTaskResult?.task?.displayTask ?? displayTaskResult?.task;
+  useEffect(() => {
+    if (task) {
+      getTests({
+        variables: {
+          taskId: task.id,
+          execution: task.execution,
+          ...(groupId && { groupId }),
+        },
+      });
+    }
+  }, [task, groupId, getTests]);
+
+  if (errorTests || errorDisplayTask) {
+    return <PageDoesNotExist />;
+  }
+
+  const { testResults } = testsResult?.taskTests ?? {};
+  return isLoadingDisplayTask || isLoadingTests ? (
+    <Skeleton paragraph={{ rows: 8 }} />
+  ) : (
+    <Row>
+      <Card>
+        <Column>
+          <>
+            <H2>
+              <StyledRouterLink to={getTaskRoute(task?.id)} data-cy="task-link">
+                {task?.id}
+              </StyledRouterLink>
+            </H2>
+            <SubtitleContainer>
+              <Subtitle>
+                Execution: <span data-cy="execution">{task?.execution}</span>
+                <br />
+                {groupId && (
+                  <>
+                    Job Number: <span data-cy="groupId">{groupId}</span>
+                  </>
+                )}
+              </Subtitle>
+            </SubtitleContainer>
+          </>
+          {testResults?.length &&
```

Comment:
only render the "Complete logs for all" link when tests exist and the given taskId in the URL is not a display task id